src/read_utils.rs

```diff
@@ -45,8 +45,8 @@ struct Config {
 pub fn read_config<P: AsRef<path::Path>>(
```

Comment:
Modification to the signature of this function is the only important part of this PR. The rest of the modification is only a consequence.